### ISSUE TITLE
linux: recent kernel fails to build

### ISSUE THREAD

**Issue by coretemp** [NixOS/nixpkgs#35849]
## Issue description
In fda4b93cd4fd3 (master) I get:
```
/tmp/nix-build-ena-1.5.0-4.15.6.drv-0/source/kernel/linux/ena/ena_netdev.c: In function 'ena_refill_rx_bufs':
/tmp/nix-build-ena-1.5.0-4.15.6.drv-0/source/kernel/linux/ena/ena_netdev.c:544:12: error: '__GFP_COLD' undeclared (first use in this function); did you mean '__GFP_COMP'?
            __GFP_COLD | GFP_ATOMIC | __GFP_COMP);
            ^~~~~~~~~~
            __GFP_COMP
compilation terminated due to -Wfatal-errors.
make[4]: *** [/nix/store/qvnq0i6mm783cqzw6z7ll7gs7szqlnjf-linux-4.15.6-dev/lib/modules/4.15.6/source/scripts/Makefile.build:317: /tmp/nix-build-ena-1.5.0-4.15.6.drv-0/source/kernel/linux/ena/ena_netdev.o] Error 1
make[3]: *** [/nix/store/qvnq0i6mm783cqzw6z7ll7gs7szqlnjf-linux-4.15.6-dev/lib/modules/4.15.6/source/Makefile:1509: _module_/tmp/nix-build-ena-1.5.0-4.15.6.drv-0/source/kernel/linux/ena] Error 2
make[2]: *** [Makefile:146: sub-make] Error 2
make[1]: *** [Makefile:24: __sub-make] Error 2
make[1]: Leaving directory '/nix/store/qvnq0i6mm783cqzw6z7ll7gs7szqlnjf-linux-4.15.6-dev/lib/modules/4.15.6/build'
make: *** [Makefile:19: all] Error 2
builder for ‘/nix/store/hdkpx9n7ygd15vff22nzwdji12nwl14w-ena-1.5.0-4.15.6.drv’ failed with exit code 2
```

### Steps to reproduce
Use `pkgs.linuxPackages_latest`

## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 4.13.4, NixOS, 17.09.git.39cd40f (Hummingbird)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.15`
 - channels(root): `"nixos-17.09.2182.7f6f0c49f0"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by dtzWill** [comment]
`pkgs.linuxPackages_latest.kernel` builds fine for me, and it doesn't appear to be the issue you're seeing.

I can't speak for `pkgs.linuxPackages_latest.ena` (where your build fails, which I can reproduce).

**Comment by dtzWill** [comment]
https://github.com/amzn/amzn-drivers/issues/56

**Comment by coretemp** [comment]
@dtzWill Thanks for that hint. The linked issue has been fixed, but I guess we just have to wait before someone fixes `ena` in nixpkgs now.

**Comment by dtzWill** [comment]
Looks like it's just a simple update, if you want to fix  it for yourself and share as a PR :).

**Comment by dotlambda** [comment]
Fixed by b353ff944a4724c09cb5aca380cce37ecc2dd5f0.
